src/org/zaproxy/zap/extension/ascanrulesAlpha/HttpOnlySite.java

```diff
@@ -0,0 +1,294 @@
+ * @author sanchitlucknow@gmail.com
+ */
+public class HttpOnlySite extends AbstractHostPlugin {
+
+	/**
+	 * Prefix for internationalised messages used by this rule
+	 */
+	private static final String MESSAGE_PREFIX = "ascanalpha.httponlysite.";
+	private static final int PLUGIN_ID = 10106;
+
+	private static final Logger log = Logger.getLogger(HttpOnlySite.class);
+	
+	@Override
+	public int getId() {
+		return PLUGIN_ID;
+	}
+
+	@Override
+	public String getName() {
+		return Constant.messages.getString(MESSAGE_PREFIX + "name");
+	}
+	
+	@Override
+	public String getDescription() {
+		return Constant.messages.getString(MESSAGE_PREFIX + "desc");
+	}
+
+	@Override
+	public String getSolution() {
+		return Constant.messages.getString(MESSAGE_PREFIX + "soln");
+	}
+
+	@Override
+	public String getReference() {
+		return Constant.messages.getString(MESSAGE_PREFIX + "refs");
+	}
+
+	@Override
+	public String[] getDependency() {
+		return null;
+	}
+
+	@Override
+	public int getCategory() {
+		return Category.MISC;
+	}
+	
+	@Override
+	public int getRisk() {
+		return Alert.RISK_LOW;
+	}
+
+	@Override
+	public int getCweId() {
+		return 311; // CWE-311: Missing Encryption of Sensitive Data
+	}
+
+	@Override
+	public int getWascId() {
+		return 4; // WASC-04: Insufficient Transport Layer Protection
+	}
+
+	@Override
+	public void init() {
+
+	}
+
+	@Override
+	public void scan() {
+		
+		if (getBaseMsg().getRequestHeader().getSecure()) { //Base request is HTTPS
+			if (log.isDebugEnabled()){
+				log.debug ("The original request was HTTPS, so there is not much point in looking further.");
+			}
+			System.out.println("\n\nBase isnt HTTP\n\n");
+			return;
+		}
+
+		if (isStop()) {
+			if (log.isDebugEnabled()) {
+				log.debug("Scanner "+getName()+" Stopping.");
+			}
+			return;
+		}
+
+		HttpMessage newRequest = getNewMsg();
+		System.out.println(newRequest.getRequestHeader().toString());
+		try{
+			log.info("**"+newRequest.getRequestHeader().getURI());
+			System.out.println(newRequest); //Checking the request
+			String host = newRequest.getRequestHeader().getURI().getHost();
+			String path = newRequest.getRequestHeader().getURI().getPath();
+			newRequest.getRequestHeader().setURI(new URI("https",null, host, 443, path));
+			log.info("**"+newRequest.getRequestHeader().getURI());
+			
+		} catch (URIException e) {
+			log.error("Error creating HTTPS URL from HTTP URL:", e);
+			return;
+		}
+
+		if (isStop()) {
+			if (log.isDebugEnabled()) {
+				log.debug("Scanner "+getName()+" Stopping.");
+			}
+			return;
+		}
+
+		try {
+			String record = "";
+			do {
+				sendAndReceive(newRequest, false);
+				System.out.println("\n\nRequest Succesful");
+				String loc = "Location: ";
+				String response = newRequest.getResponseHeader().toString();
+				System.out.println(response);
+				int ind = response.indexOf(loc);
+				String found = response.substring(ind+10, response.length());
+				System.out.println(ind+" Index");
+				if(ind > 0) {
```

Comment:
It's better to use the status code to know if the response is a redirection. [1] [2]
Also worth checking/using the class HttpStatusCode. [3]

[1] https://github.com/zaproxy/zaproxy/blob/dc1ebd5d9af75cd8067c374058dcf153505a844a/src/org/parosproxy/paros/network/HttpResponseHeader.java#L104
[2] http://tools.ietf.org/html/rfc7231#section-6.4
[3] https://github.com/zaproxy/zaproxy/blob/cfe9796a8c1b1377d52d6be3142a22a02d1203eb/src/org/parosproxy/paros/network/HttpStatusCode.java